iceoryx2-bb/cxx/include/iox2/bb/into.hpp

```diff
@@ -27,19 +30,34 @@ namespace detail {
 /// 'into<lossy<T>>'
 template <typename T>
 struct extract_into_type {
-    using type_t = T;
+    using TargetType = T;
 };
 
 /// @brief Helper struct to get the actual destination type 'T' for 'into<lossy<T>>'
 template <typename T>
 struct extract_into_type<lossy<T>> {
-    using type_t = T;
+    using TargetType = T;
 };
 } // namespace detail
 
+// Using a struct as impl, as free functions do not support partially specialized templates
+template <typename SourceType, typename DestinationType>
+struct FromTrait {
```

Comment:
Renamed `FromImpl` to `FromTrait`. `Impl` was not quite the right name. As alternative to `FromTrait`, just `From` would also work well.

Reply:
I would vote for just `From`